src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -309,17 +309,25 @@ void GcsActorManager::HandleGetActorInfo(rpc::GetActorInfoRequest request,
                  << ", job id = " << actor_id.JobId() << ", actor id = " << actor_id;
 
   const auto &registered_actor_iter = registered_actors_.find(actor_id);
+  std::shared_ptr<GcsActor> *ptr = nullptr;
+  auto arena = reply->GetArena();
+  RAY_CHECK(arena != nullptr);
   if (registered_actor_iter != registered_actors_.end()) {
-    reply->unsafe_arena_set_allocated_actor_table_data(
-        registered_actor_iter->second->GetMutableActorTableData());
+    ptr = google::protobuf::Arena::Create<std::shared_ptr<GcsActor>>(
+        arena, registered_actor_iter->second);
   } else {
     const auto &destroyed_actor_iter = destroyed_actors_.find(actor_id);
     if (destroyed_actor_iter != destroyed_actors_.end()) {
-      reply->unsafe_arena_set_allocated_actor_table_data(
```

Comment:
This is unrelated to the PR right? Is it to use Arena on actor related RPCs?

Reply:
No, it is related. See here we send Finish in another thread which means these data needs to be alive until they got sent.
Here arena::create with a placement new it'll create a shared ptr so that it's shared across threads.

Otherwise, it's going to crash.